### ISSUE TITLE
override vterm_unicode_width with utf_char2cells

### ISSUE THREAD

**Issue by mattn** [vim/vim#2006]
Some unicode characters are displayed as two cells in vim but there are displayed as one cell in `:terminal`. Below is workaround patch to fix this issue. But I wonder whether we should apply patches for libvterm.

@brammool  how do you think?

```diff
diff --git a/src/libvterm/src/unicode.c b/src/libvterm/src/unicode.c
index b989add3c..8383f67e6 100644
--- a/src/libvterm/src/unicode.c
+++ b/src/libvterm/src/unicode.c
@@ -181,6 +181,7 @@ static int bisearch(uint32_t ucs, const struct interval *table, int max) {
  */
 
 
+#if 0
 static int mk_wcwidth(uint32_t ucs)
 {
   /* test for 8-bit control characters */
@@ -211,6 +212,7 @@ static int mk_wcwidth(uint32_t ucs)
       (ucs >= 0x20000 && ucs <= 0x2fffd) ||
       (ucs >= 0x30000 && ucs <= 0x3fffd)));
 }
+#endif
 
 #if 0 /* unused */
 static int mk_wcswidth(const uint32_t *pwcs, size_t n)
@@ -322,7 +324,8 @@ static int mk_wcswidth_cjk(const uint32_t *pwcs, size_t n)
 
 INTERNAL int vterm_unicode_width(uint32_t codepoint)
 {
-  return mk_wcwidth(codepoint);
+  extern int utf_char2cells(int);
+  return utf_char2cells((int) codepoint);
 }
 
 INTERNAL int vterm_unicode_is_combining(uint32_t codepoint)
```

**Comment by brammool** [comment]
Yasuhiro Matsumoto wrote:

> Some unicode characters are displayed as two cells in vim but there
> are displayed as one cell in `:terminal`. Below is workaround patch to
> fix this issue. But I wonder whether we should apply patches for
> libvterm.
> 
> @brammool  how do you think?

Thanks for the idea. I think we should do the same as with vsnprintf().

-- 
A consultant is a person who takes your money and annoys your employees while
tirelessly searching for the best way to extend the consulting contract.
				(Scott Adams - The Dilbert principle)

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///

**Comment by k-takata** [comment]
Fixed by [8.0.0985](https://github.com/vim/vim/commit/6d0826dfbba9880820d9ec221327e4250bbf6540).
